owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -360,10 +360,15 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
                     Timber.d("Service discovery: ${it.peekContent().getStoredData()}")
                     oidcSupported = true
                     it.peekContent().getStoredData()?.let { oidcServerConfiguration ->
-                        registerClient(
-                            oidcServerConfiguration.authorizationEndpoint.toUri(),
-                            oidcServerConfiguration.registrationEndpoint
-                        )
+                        val registrationEndpoint = oidcServerConfiguration.registrationEndpoint
+                        if (registrationEndpoint != null) {
+                            registerClient(
+                                oidcServerConfiguration.authorizationEndpoint.toUri(),
+                                registrationEndpoint
+                            )
+                        } else {
+                            performGetAuthorizationCodeRequest(oidcServerConfiguration.authorizationEndpoint.toUri())
+                        }
```

Comment:
How about using `let`and `?:` to kotlinize it even more? Just a suggestion, no important changes
```suggestion
                        registrationEndpoint?.let {
                            registerClient(
                                oidcServerConfiguration.authorizationEndpoint.toUri(),
                                registrationEndpoint
                            ) ?: performGetAuthorizationCodeRequest(oidcServerConfiguration.authorizationEndpoint.toUri())
```

Reply:
Yes, of course, we could do it that way. I used the `if else` because from my point of view it is more clear than `let {} ?:` but it is basically the same.